uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar-month.component.spec.js

```diff
@@ -0,0 +1,695 @@
+    'use strict';
+
+    describe('leaveCalendarMonth', function () {
+      var $componentController, $log, $provide, $q, $rootScope, Calendar,
+        LeaveRequest, OptionGroup, controller, daysInFebruary, february, leaveRequestInFebruary,
+        period2016, publicHolidays;
+      var currentContactId = CRM.vars.leaveAndAbsences.contactId;
+
+      beforeEach(module('leave-absences.templates', 'leave-absences.mocks', 'my-leave', function (_$provide_) {
+        $provide = _$provide_;
+      }));
+
+      beforeEach(inject(function (LeaveRequestAPIMock, WorkPatternAPIMock) {
+        $provide.value('LeaveRequestAPI', LeaveRequestAPIMock);
+        $provide.value('WorkPatternAPI', WorkPatternAPIMock);
+      }));
+
+      beforeEach(inject(function (_$componentController_, _$log_, _$q_, _$rootScope_,
+        _Calendar_, _LeaveRequest_, _OptionGroup_, OptionGroupAPIMock) {
+        $componentController = _$componentController_;
+        $log = _$log_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        Calendar = _Calendar_;
+        LeaveRequest = _LeaveRequest_;
+        OptionGroup = _OptionGroup_;
+
+        february = { index: 1, year: 2016 };
+        daysInFebruary = moment().month(february.index).year(february.year).daysInMonth();
+        period2016 = _.clone(AbsencePeriodData.all().values[0]);
+        publicHolidays = PublicHolidayData.all().values;
+        leaveRequestInFebruary = LeaveRequestData.all().values[0];
+
+        spyOn($log, 'debug');
+        spyOn(Calendar, 'get').and.callThrough();
+        spyOn(LeaveRequest, 'all').and.callFake(function () {
+          return $q.resolve({ list: [leaveRequestInFebruary] }); // leave request from February
+        });
+        spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+          return OptionGroupAPIMock.valuesOf(name);
+        });
+
+        compileComponent();
+      }));
+
+      it('is initialized', function () {
+        expect($log.debug).toHaveBeenCalled();
+      });
+
+      it('not showing', function () {
+        expect(controller.visible).toBe(false);
+      });
+
+      it('has the maximum size of the contacts list per page defined', function () {
+        expect(controller.pageSize).toBeDefined();
+        expect(controller.pageSize).toEqual(jasmine.any(Number));
+      });
+
+      it('has the current page set to zero', function () {
+        expect(controller.currentPage).toBe(0);
+      });
+
+      it('has the show-contact-name binding set to false by default', function () {
+        expect(controller.showContactName).toBe(false);
+      });
+
+      it('has the show-only-with-leave-request name binding set to false by default', function () {
+        expect(controller.showOnlyWithLeaveRequests).toBe(false);
+      });
+
+      describe('on init', function () {
+        it('does not load the contacts\' work pattern calendars', function () {
```

Comment:
Please avoid using ' because escaping does not look good. `does not load work pattern calendars of the contacts`. Same for other places in the code.